src/main/java/com/cdancy/bitbucket/rest/domain/branch/Type.java

```diff
@@ -17,24 +17,41 @@
 
 package com.cdancy.bitbucket.rest.domain.branch;
 
+import org.jclouds.javax.annotation.Nullable;
 import org.jclouds.json.SerializedNames;
 
 import com.google.auto.value.AutoValue;
 
 @AutoValue
 public abstract class Type {
 
-    public abstract String id();
+    public enum TypeId {
+        BUGFIX,
+        FEATURE,
+        HOTFIX,
+        RELEASE
+    }
+
+    public abstract TypeId id();
 
+    @Nullable
     public abstract String displayName();
 
     public abstract String prefix();
 
+    @Nullable
+    public abstract Boolean enabled();
```

Comment:
Can this really be Null? If not lets use unboxed boolean here and below in the `create` method.